### PR TITLE
Feature/binaries from aaf s3

### PR DESCRIPTION
**Pull request by trsau**
All external binaries load from AAF's S3 and version number at top of site.xml

### CODE REVIEW

File: site.yml [hunk 1]
```diff
@@ -43,17 +67,20 @@
       username: idp_admin
       password: "{{ lookup('password', installer.repository + '/passwords/' + inventory_hostname + '/mariadb chars=letters,digits length=64') }}"
       mysql_connector:
-        version: 5.1.35
-        sha256sum: fa77909a62363c1aa49ef5ab537ed4affe90a4a6aa5dd1045555b16b2611306d
```
Comment (bradleybeddoes) [inline site.yml]: No sha256sum ?

File: site.yml [hunk 2]
```diff
@@ -43,17 +67,20 @@
...
+        version: "{{ software.mysql_connector.version }}"
+        url: "{{ software.mysql_connector.baseurl }}/mysql-connector-java-{{ software.mysql_connector.version }}.tar.gz"
     aaf_shib_ext:
-      version: 1.0.1
+      version: "{{ software.aaf_shib_ext.version }}"
+      url : "{{ software.aaf_shib_ext.baseurl }}/aaf-shib-ext-{{ software.aaf_shib_ext.version }}.jar"
```
Comment (bradleybeddoes) [inline site.yml]: No sha256sum here either?

File: site.yml [hunk 3]
```diff
@@ -9,8 +9,30 @@
...
+        baseurl: "{{ aaf_binaries.baseurl }}/jars/commons_dbcp2"
+        version: 2.1.1
+        sha256sum: d6181397b20490aa9b3a793a19c2b75f31af032040d976db1cd97cc2f0ef4379
+      commons_pool2:
+        baseurl: "{{ aaf_binaries.baseurl }}/jars/commons_pool2"
+        version: 2.4.2
+        sha256sum: 175ddacc9364b4619ac0fd0fd22acc9d4c372f9099ae5208e8cb72e04a9a872e
+      aaf_shib_ext:
+        baseurl: "{{ aaf_binaries.baseurl }}/jars/aaf_shib_ext"
+        version: 1.0.1
```
Comment (bradleybeddoes) [inline site.yml]: No sha256sum ?

**Comment by bradleybeddoes** [comment]
Ok I think I see what is happening here.

Sometimes `get_url` is referencing sha256sum values like so `'{{ software.mysql_connector.sha256sum }}'`and at other times like so `sha256sum: '{{ shib_idp.sha256sum }}'`.

Hence it looks to me like some sha256sum values are missing within `site.yml` when in reality they are just being referenced differently.

I think consideration needs to be given so that we're always referencing these values in the same way within tasks. Thoughts?

I also noted some usage of `get_url` where no `sha256sum` exists at all, e.g. for tasks/idp.yml 'Download placeholder keystore' (which incidentally probably needs to be considered a binary and hosted on S3 if we're going to rely on it). I'd suggest a quick check over all `get_url` calls might be of benefit to weed out anything else not being correctly validated.

**Comment by trsau** [comment]
The original had bits all over the place, so first pass is to consolidate all of the version details and sha256sum in one place - top of the site.xml file in the software section. I need to add a sha256sum to the aaf_shib_ext as it never had one :( 

Next pass is the ensure a consistent way of using these values.  Will also look for any other get_url call and ensure they are consistent. Will do as a new feature.

**Comment by trsau** [comment]
@bradleybeddoes All get_urls are done the same consistent way (apart from the Metadata certificate that will be done when I move to the new MD source). All references to files urls, versions, and sha256sum values are also consistent across the playbook.

Tested with a new install and upgrade.

**Comment by bradleybeddoes** [comment]
Awesome :+1:

**Pull request merged**